pool.go

```diff
@@ -39,24 +43,62 @@ func (p *pool) loadPool() ([]string, error) {
 	return responses, nil
 }
 
+type updateRingMembersReq struct {
+	newMembers []string
+	done       chan struct{}
+}
+
+type getRingNodesReq struct {
+	key  string
+	n    int
+	resp chan getRingNodesResp
+}
```

Comment:
I'm worried about the scaling limitation that would happen if standard gets are synchronized this way.

in the current design, the standard case is that each inbound request takes an `RLock`, which many threads can do without a point of synchronization/serialization. By making every in-bound request ask for nodes from a single event-loop thread you're serializing all requests against what a single core / event-loop go-routine has time for.